src/Exception/AuthenticationException.php

```diff
@@ -13,12 +13,11 @@
 
 class AuthenticationException extends \RuntimeException implements ApplicationRelatedException
 {
-    private $applicationCode;
-
-    public function __construct(string $applicationCode, string $message = '', \Throwable $previous = null)
-    {
-        $this->applicationCode = $applicationCode;
-
+    public function __construct(
+        private string $applicationCode,
+        string $message = '',
+        \Throwable $previous = null
+    ) {
         parent::__construct($message, 0, $previous);
     }
```

Comment:
I'm not super fan of mixing class property and argument in the constructor.
But don't change it now, let's wait for more feedback

Reply:
Not a big deal IMO. I think I prefer consistency between all property declarations